src/test/java/teammates/test/cases/testdriver/TestngTest.java

```diff
@@ -0,0 +1,115 @@
+package teammates.test.cases.testdriver;
+
+import static org.testng.AssertJUnit.assertTrue;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.util.HashMap;
+import java.util.Map.Entry;
+
+import org.testng.annotations.Test;
+
+import teammates.common.util.FileHelper;
+import teammates.test.cases.BaseTestCase;
+
+public class TestngTest extends BaseTestCase {
+
+    @Test
+    public void checkTestsInTestng() throws FileNotFoundException {        
+        String testNgXml = getFileContent("./src/test/testng.xml");
+        HashMap<String, String> testFiles = getTestFiles(testNgXml); // <class name, package name>
+        
+        testFiles = excludeFiles(testFiles);
+        
+        for (Entry<String, String> testFileName : testFiles.entrySet()) {
+            assertTrue(isTestFileIncluded(testNgXml, testFileName));
+        }
+    }
+    
+    private String getFileContent(String file) throws FileNotFoundException {
```

Comment:
It doesn't improve SLAP. My point is the extra method does not improve SLAP either so might as well not have it. i.e. we can get rid of the extra method call without reducing SLAP.